### PR TITLE
simsensors: support for GPS on yasim

### PR DESCRIPTION
**Pull request by mlyle**
Yasim remains WIP, but this is an important step.  Salvaged from my 'fixedwingsim' branch.

### CODE REVIEW

**Comment by mlyle** [comment]
uhhh

jenkins, test this please

**Comment by mlyle** [comment]
jenkins, test this please

File: flight/Modules/Stabilization/simulated/simsensors.c [hunk 1]
```diff
@@ -570,21 +571,69 @@ static void simulateYasim()
 		last_baro_time = PIOS_DELAY_GetRaw();
 	}
 
+	// Update GPS periodically
+	static uint32_t last_gps_time = 0;
+	static float gps_vel_drift[3] = {0,0,0};
+	if(PIOS_DELAY_DiffuS(last_gps_time) / 1.0e6 > GPS_PERIOD) {
```
Comment (tracernz) [inline flight/Modules/Stabilization/simulated/simsensors.c:577]: Space

File: flight/Modules/Stabilization/simulated/simsensors.c [hunk 2]
```diff
@@ -570,21 +571,69 @@ static void simulateYasim()
 		last_baro_time = PIOS_DELAY_GetRaw();
 	}
 
+	// Update GPS periodically
+	static uint32_t last_gps_time = 0;
+	static float gps_vel_drift[3] = {0,0,0};
```
Comment (tracernz) [inline flight/Modules/Stabilization/simulated/simsensors.c:576]: Spaces ,

**Comment by mlyle** [comment]
jenkins, test this please

File: flight/Modules/Stabilization/simulated/simsensors.c [hunk 3]
```diff
@@ -570,21 +571,69 @@ static void simulateYasim()
...
+	if(PIOS_DELAY_DiffuS(last_gps_time) / 1.0e6 > GPS_PERIOD) {
+		// Use double precision here as simulating what GPS produces
+		double T[3];
+		T[0] = homeLocation.Altitude+6.378137E6f * DEG2RAD;
+		T[1] = cosf(homeLocation.Latitude / 10e6 * DEG2RAD)*(homeLocation.Altitude+6.378137E6) * DEG2RAD;
```
Comment (tracernz) [inline flight/Modules/Stabilization/simulated/simsensors.c:581]: Spaces around operators. What is 6.378137E6f ? And the second copy of it doesn't have single precision suffix.

File: flight/Modules/Stabilization/simulated/simsensors.c [hunk 4]
```diff
@@ -570,21 +571,69 @@ static void simulateYasim()
...
+		gps_drift[0] = gps_drift[0] * 0.95 + rand_gauss() / 10.0;
+		gps_drift[1] = gps_drift[1] * 0.95 + rand_gauss() / 10.0;
+		gps_drift[2] = gps_drift[2] * 0.95 + rand_gauss() / 10.0;
+
+		GPSPositionData gpsPosition;
+		GPSPositionGet(&gpsPosition);
+		gpsPosition.Latitude =  (status.lat + gps_drift[0] / T[0]) * 10.0e6;
+		gpsPosition.Longitude = (status.lon + gps_drift[1] / T[1]) * 10.0e6;
+		gpsPosition.Altitude =  (status.alt + gps_drift[2]) / T[2];
+		gpsPosition.Groundspeed = sqrtf(pow(status.vel[0] + gps_vel_drift[0],2) + pow(status.vel[1] + gps_vel_drift[1],2));
```
Comment (tracernz) [inline flight/Modules/Stabilization/simulated/simsensors.c:594]: Spaces

File: flight/Modules/Stabilization/simulated/simsensors.c (hunk 3 above)
Comment (mlyle) [reply flight/Modules/Stabilization/simulated/simsensors.c:581]: I assume it's the distance to the middle of the earth, from MSL.. Though this is just factored from other code (and I dind't change spacing from the existing code)

Comment (tracernz) [reply flight/Modules/Stabilization/simulated/simsensors.c:581]: It should be a define. Things like that shouldn't need to be looked up to read code. :/

**Comment by tracernz** [comment]
As per IRC, sub-par code is actually already in tree in multiple places. If it's not all going to be fixed or factored out immediately, it makes sense to keep it the same so it can more easily be mass refactored later. :/ I don't like merging stuff with issues but it seems the lesser evil.

**Pull request merged**